PovertyAction/ipa-data-tech-handbook
# Justfile and README

## Pull Request Summary 🚀

### What does this PR do? 📝

This PR initializes the README instructions for setting up a local computing environment with python and quarto to write and build the data and tech handbook.

In this review, I would like the following feedback:

- Do the tests below run successfully?
- Are there any usage difficulties you encountered while installing the software and building the local virtual environment?

### Why is this change needed? 🤔


This is the first PR to begin the scaffolding for the data, engineering, and technology handbook. The changes are needed in order to set a base set of software needed to develop, preview, and build the handbook.

### How was this implemented? 🛠️


I added the required Python packages to `pyproject.toml` using `rye add ...` and then tested the Python virtual environment using `rye sync`.

I then built the initial handbook with just a few pages in `_quarto.yml`:

```yaml
    right:
      - href: ./handbook/index.qmd
        text: Home
      - href: ./handbook/computer_setup.qmd
        text: Setup
```

I created the Justfile with helpful aliases for common patterns:

<img width="486" alt="image" src="https://github.com/PovertyAction/ipa_data_tech_handbook/assets/5607589/eceb3893-0ab9-4843-8d01-383ce1b3f3f1">

### How to test or reproduce changes? 🧪


Pre-requisites (See the README)
- make sure that you have git and Just installed on your computer and have a Terminal Shell to work from. You may also want GitHub Desktop.

```bash
winget install Git.Git Casey.Just GitHub.GithubDesktop
```

As a part of this review, I am requesting the following:

1. Go through the README.md instructions to install required software and build the local virtual environment. Specifically, run to ensure the python virtual environment and quarto are installed on your computer.

```bash
just get-started
```

2. Preview the initial pages of the data and technology handbook (this is a work in progress, however, I am looking for confirmation that the following runs properly and previews the handbook in a browser window on your laptop.

```bash
just preview-docs
```

After running that last command you should see the handbook in a browser window. There are currently two pages

*Home*
<img width="1826" alt="image" src="https://github.com/PovertyAction/ipa_data_tech_handbook/assets/5607589/ead35f04-a86e-4e2b-8d5c-0d7e24dae48b">

*Setup*
<img width="1829" alt="image" src="https://github.com/PovertyAction/ipa_data_tech_handbook/assets/5607589/77e91dfe-9095-4641-bd3a-501d3017a8b7">




### Screenshots (if applicable) 📷


### Checklist ✅
